artman/tasks/protoc_tasks.py

```diff
@@ -120,33 +141,50 @@ class ProtoCodeGenTask(ProtocCodeGenTaskBase):
     """Generates protos"""
     def execute(self, language, src_proto_path, import_proto_path,
                 output_dir, api_name, api_version, organization_name,
-                toolkit_path, gapic_yaml, final_src_proto_path=None,
+                toolkit_path, gapic_yaml, root_dir, final_src_proto_path=None,
                 final_import_proto_path=None, excluded_proto_path=[]):
         pkg_dir = protoc_utils.prepare_proto_pkg_dir(
             output_dir, api_name, api_version, organization_name, language)
         return self._execute_proto_codegen(
             language, src_proto_path, import_proto_path, pkg_dir,
             api_name, api_version, organization_name, toolkit_path,
-            gapic_yaml, gen_proto=True,
+            gapic_yaml, root_dir, gen_proto=True,
             final_src_proto_path=final_src_proto_path,
             final_import_proto_path=final_import_proto_path,
             excluded_proto_path=excluded_proto_path)
 
+class ResourceNameGenTask(ProtocCodeGenTaskBase):
```

Comment:
Just curious: what do you have to add this task? Shouldn't artman already be able to invoke the resource name plugin and generate resource name classes?

Reply:
I had to split generation of resource names and generation of protobuf stubs. Since now plugin does not use insertion poiints, nothing prevents us from doing it. I had to do it to prevent generation of proto stubs for the common resource names messages - since to declare annotation we need a message, `common_resources.proto` declares those "fake" messages which exist solely to declare resource annotation in them.

Reply:
That makes sense.